aspnetcore/mvc/razor-pages/ui-class.md

```diff
@@ -0,0 +1,217 @@
+---
+title: Reusable Razor UI in class libraries with ASP.NET Core
+author: Rick-Anderson
+description: Explains how to create reusable Razor Pages UI in a class library.
+manager: wpickett
+monikerRange: '>= aspnetcore-2.1'
+ms.author: riande
+ms.date: 4/31/2018
+ms.prod: asp.net-core
+ms.technology: aspnet
+ms.topic: advanced
+uid: mvc/razor-pages/ui-class
+---
+# Create reusable UI using the Razor Class Library project.
+
+By [Rick Anderson](https://twitter.com/RickAndMSFT)
+
+Razor views, pages, controllers, page models, and data models can be built into a Razor Class Library(RCL). The RCL can be and packaged and reused. Applications can include the RCL and override the views and pages it contains. When a view, partial view, or Razor Page is found in both the web app and the RCL, the Razor markup (*.cshtml* file) in the web app takes precedence.
+
+This feature requires [!INCLUDE[](~/includes/2.1-SDK.md)]
+
+[!INCLUDE[](~/includes/2.1.md)]
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/mvc/razor-pages/ui-class/samples) ([how to download](xref:tutorials/index#how-to-download-a-sample))
+
+## Create a class library containing Razor UI
+
+# [Visual Studio](#tab/visual-studio)
+
+* From the Visual Studio **File** menu, select **New** > **Project**.
+* Select **ASP.NET Core Web Application**.
+* Verify **ASP.NET Core 2.1** or later is selected.
+* Select **Razor Class Library** > **OK**.
+
+# [.NET Core CLI](#tab/netcore-cli)
+
+From the commandline, run `dotnet new razorclasslib`. For example:
+
+``` CLI
+dotnet new razorclasslib -o RazorUIClassLib
+```
+
+For more information, see [dotnet new](/dotnet/core/tools/dotnet-new).
+
+------
+Add Razor files to the RCL.
+
+## Razor Class Library restrictions.
+
+* Razor files must be under the *Areas* folder.
```

Comment:
This isn't a requirement, we just think using Areas is a better model for sharing isolated pieces of UI libraries.